hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -239,6 +245,30 @@ else if ( useJdbcMetadata ) {
 				log.unableToObtainConnectionToQueryMetadata( e );
 			}
 		}
+		else {
+			if (
+				(StringHelper.isNotEmpty( explicitDatabaseVersion ) || explicitDatabaseMajorVersion != null || explicitDatabaseMinorVersion != null)
+				&& ( StringHelper.isNotEmpty( explicitDatabaseName ) || StringHelper.isNotEmpty( (String) configurationValues.get( AvailableSettings.DIALECT ) ) )
+			) {
+				return new JdbcEnvironmentImpl(
+						registry,
+						dialectFactory.buildDialect(
+								configurationValues,
+								() -> new DialectResolutionInfoImpl(
```

Comment:
Create a new variable for `DialectResolutionInfoImpl` to avoid the lambda issue: https://github.com/hibernate/hibernate-orm/pull/5206#discussion_r966116210